File: FibonacciChecker.java
import java.util.Scanner;

public class FibonacciChecker {

    public static boolean isFibonacci(int numero) {
        int a = 0, b = 1;
        
            if (numero == 0) return true;

            while (b <= numero) {
                if (b == numero) {
                    return true; 
            }

            int temp = a;
            a = b;
            b = temp + b;  
        }

        return false; 
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Informe um número: ");
        int numero = scanner.nextInt();

        if (isFibonacci(numero)) {
            System.out.println("O número " + numero + " pertence à sequência de Fibonacci.");
        }else{
            System.out.println("O número " + numero + " NÃO pertence à sequência de Fibonacci.");
        }

        scanner.close();
    }
}

File: FibonacciChecker.cs
using System;

class Program
{
    static void Main(string[] args)
    {

        Console.Write("Informe um número: ");
        if (int.TryParse(Console.ReadLine(), out int numero))

        {
            if (VerificaFibonacci(numero))

            {

                Console.WriteLine($"{numero} pertence à sequência de Fibonacci.");

            }else{

                Console.WriteLine($"{numero} NÃO pertence à sequência de Fibonacci.");

            }
        }else{

            Console.WriteLine("Por favor, informe um número válido.");

        }
    }

    static bool VerificaFibonacci(int numero)

    {
        if (numero < 0) return false;

        int a = 0, b = 1;

        while (a < numero)
        {
            int temp = a;
            a = b;
            b = temp + b;
        }

        return a == numero;
    }
}
